vault/cors.go

```diff
@@ -0,0 +1,112 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+var responseHeaders = map[string]string{
+	"Access-Control-Allow-Headers":     "origin,content-type,cache-control,accept,options,authorization,x-requested-with,x-vault-token",
+	"Access-Control-Max-Age":           "1800",
+	"Access-Control-Allow-Credentials": "true",
+	"Vary": "Origin",
+}
+
+var allowedMethods = []string{
+	http.MethodDelete,
+	http.MethodGet,
+	http.MethodOptions,
+	http.MethodPost,
+	http.MethodPut,
```

Comment:
I've added `LIST`. These methods are the methods that the Vault API accepts. Setting it to `*` would allow requests for methods that Vault does not accept (PATCH for example) , then the actual request would be sent and Vault would respond with a 404. 

Leaving this as a list of acceptable methods means that Vault will respond to preflight request for an unacceptable method in a way that will cause the browser to report a meaningful error and will prevent the actual request from being sent. Since preflight responses are cached on the client-side this will also prevent repeated unacceptable requests from being sent.